MISRA.md

```diff
@@ -0,0 +1,37 @@
+| Directive 4.8 | Advisory | Allow inclusion of unused types. Header files for a specific port, which are needed by all files, may define types that are not used by a specific file. |
+| Directive 4.9 | Advisory | Allow inclusion of function like macros. The `assert` macro is used throughout the library for parameter validation, and logging is done using function like macros. |
+| Rule 2.3 | Advisory | Allow unused types. Library headers may define types intended for the application's use, but not used within the library files. |
+| Rule 2.4 | Advisory | Allow unused tags. Some compilers warn if types are not tagged. |
+| Rule 2.5 | Advisory | Allow unused macros. Library headers may define macros intended for the application's use, but are not used by a specific file. |
+| Rule 3.1 | Required | Allow nested comments. C++ style `//` comments are used in example code within Doxygen documentation blocks. |
+| Rule 11.5 | Advisory | Allow casts from `void *`. Fields may be passed as `void *`, requiring a cast to the correct data type before use. |
+| Rule 21.1 | Required | Allow use of all macro names. For compatibility, libraries may define macros introduced in C99 for use with C90 compilers. |
+| Rule 21.2 | Required | Allow use of all macro and identifier names. For compatibility, libraries may define macros introduced in C99 for use with C90 compilers. |
+
+### Flagged by Coverity
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Rule 5.7 | Required | The network context struct is defined by each transport implementation; however, only one transport implementation is needed by an application at a time. If the transport implementations are analyzed one at a time by Coverity, this violation will not be flagged. |
+| Rule 8.7 | Advisory | API functions are not used by the library outside of the files they are defined; however, they must be externally visible in order to be used by an application. |
+
+### Suppressed with Coverity Comments
+| Deviation | Category | Justification |
+| :-: | :-: | :-: |
+| Directive 4.6 | Advisory | Basic numerical types are used in platform implementations, since the POSIX platform functions and openssl functions used in the platform implementations take arguments of these types. |
+| Rule 10.1 | Required | A POSIX specific macro utility `FD_SET` is flagged for this violation. This macro utility, whose implementations is supplied by the system, is used in the transport implementation. |
+| Rule 10.8 | Required | A POSIX specific macro utility `FD_SET` is flagged for this violation. This macro utility, whose implementations is supplied by the system, is used in the transport implementation. |
+| Rule 11.3 | Required | The transport implementation casts an object pointer to a pointer of a different object type. This cast is supported in POSIX, and is used to obtain the IP address from an address record. |
```

Comment:
I see you changed `an` back to `the`. I didn't like `the` since it implies we've referenced it before. Here's another suggestion:
```suggestion
| Rule 11.3 | Required | The transport implementation casts an object pointer to a pointer of a different object type. This cast is supported in POSIX, and is used to obtain IP addresses from address records. |
```

Reply:
I didn't like the 'an' there. The new suggestion is good. Thanks.